metis-reprocessing/src/main/java/eu/europeana/metis/reprocessing/utilities/IndexUtilities.java

```diff
@@ -56,15 +104,275 @@ public static void indexRecord(RDF rdf, Boolean preserveTimestamps, Configuratio
+                                         .map(Arrays::asList).flatMap(List::stream).toList();
+
+    return fullBean.getAggregations().stream().filter(x -> proxyInResult.contains(x.getAbout())).toList();
+  }
+
+  private static Pair<Set<String>, Map<String, List<String>>> extractUrisAndLiterals(
+      final Map<String, List<String>> urisLiteralsMap, Map<String, Pair<String, String>> organizationPrefLabelMap) {
+    final Set<String> organizationUris = new HashSet<>();
+    final Map<String, List<String>> literalsMap = new HashMap<>();
+
+    if (MapUtils.isNotEmpty(urisLiteralsMap)) {
+      splitOrganizationUrisFromLiterals(urisLiteralsMap, organizationUris, literalsMap, organizationPrefLabelMap);
+
+      //Extend map with organization pref labels
+      if (CollectionUtils.isNotEmpty(organizationUris)) {
+        addOrganizationPrefLabelsToLiterals(organizationUris, literalsMap, organizationPrefLabelMap);
+      }
+    }
+    return new ImmutablePair<>(organizationUris, literalsMap);
+  }
+
+  private static void splitOrganizationUrisFromLiterals(Map<String, List<String>> urisLiteralsMap,
+      Set<String> organizationUris, Map<String, List<String>> literalsMap,
+      Map<String, Pair<String, String>> organizationPrefLabelMap) {
+    for (Map.Entry<String, List<String>> entry : urisLiteralsMap.entrySet()) {
+      final List<String> literals = new ArrayList<>();
+      for (String value : entry.getValue()) {
+        if (organizationPrefLabelMap.containsKey(value)) {
+          organizationUris.add(value);
+        } else {
+          literals.add(value);
+        }
+      }
+      if (!literals.isEmpty()) {
+        literalsMap.put(entry.getKey(), literals);
+      }
+    }
+  }
+
+  private static void addOrganizationPrefLabelsToLiterals(Set<String> organizationUris,
+      Map<String, List<String>> literalsMap, Map<String, Pair<String, String>> organizationPrefLabelMap) {
+    for (String organizationUri : organizationUris) {
+      final Pair<String, String> entry = organizationPrefLabelMap.get(organizationUri);
+      if (entry != null) {
+        literalsMap.computeIfAbsent(entry.getKey(), key -> new ArrayList<>()).add(entry.getValue());
+      }
+    }
+  }
+
+  private static String getDatasetIdOfRecordToBePurged(RDF rdf) {
+    Optional<String> about = rdf.getProvidedCHOList()
+                                .stream()
+                                .filter(Objects::nonNull)
+                                .findFirst()
+                                .map(ProvidedCHOType::getAbout);
+
+    String result = "";
+    if (about.isPresent()) {
+      String datasetId = about.get().substring(1, StringUtils.ordinalIndexOf(about.get(), "/", 2));
+      if (datasetId.equals("9200359") || datasetId.equals("9200579")
+          || datasetId.equals("2048128") || datasetId.equals("2048087")) {
+        result = datasetId;
+      }
+    }
+    return result;
+  }
+
+  private static boolean isProviderProxy(ProxyType proxy) {
```

Comment:
Some of those are in RdfWrapper class, maybe useful.
Just noting this, it is not very important generally since this is code that we change on each iteration.